gh-117151: IO performance improvement, increase io.DEFAULT_BUFFER_SIZE to 128k (#118144)
python/cpython: Misc/NEWS.d/next/Core and Builtins/2024-04-30-14-03-09.gh-issue-117151.yt2H8c.rst

> increase io.DEFAULT_BUFFER_SIZE from 4k/8k to 128k. fix open on Linux to use

Reviewer: ```suggestion
Increase ``io.DEFAULT_BUFFER_SIZE`` from 4k/8k to 128k. Adapt :func:`open` on Linux to use
```

> Increase ``io.DEFAULT_BUFFER_SIZE`` from 4k/8k to 128k. Adapt :func:`open` on Linux to use

Reviewer 2: I think that this should be rather in the "Library" section than in "Core and Builtins".

Author: This was set here by the news tool automatically. I think "Core and Builtins" makes more sense for the I/O buffering code to be considered core and builtins, though I can move if you prefer?

Reviewer 2: While `open()` is exposed as a builtin, it is defined in the `io` module. This is not a part of the core interpreter.

> max(io.DEFAULT_BUFFER_SIZE, device block size) rather than the device block

Reviewer: ```suggestion
``max(io.DEFAULT_BUFFER_SIZE, device block size)`` rather than the device block
```

> size. improve I/O performance by 3 to 5 times. patch by Romain Morotti.

Reviewer: ```suggestion
size. Improve I/O performance upto 3 to 5 times. Patch by Romain Morotti.
```
